### ISSUE TITLE
Rebuild database with partitioned tables

### ISSUE THREAD

**Issue by znmeb** [hackoregon/transportation-congestion-analysis#17]
This is a partial fix for https://github.com/hackoregon/2019-transportation-systems-backend/issues/30. See https://stackoverflow.com/questions/49776856/postgresql-multicolumn-indexes-vs-single-column-index and https://www.postgresql.org/docs/11/ddl-partitioning.html

**Comment by znmeb** [comment]
Status update:

There are two pieces to the implementation:
1. I've rewritten the "extract" portion of the database creation. This is the step that unpacks the CSV files from the archives. The previous version just extracted them and all the logic around different data formats and column selection was done later, in PostgreSQL.

    The current version pushes all this logic into the extraction process, using a command-line tool called `xsv` (https://github.com/BurntSushi/xsv). There is now one CSV for each table type for each month, only the columns used are in the files, and the dates have the same format in all of them.

2. The large tables are partitioned by month. I'm using the simplest partitioning scheme, declarative partitioning (https://www.postgresql.org/docs/11/ddl-partitioning.html#DDL-PARTITIONING-DECLARATIVE). The resulting SQL is much easier to read because the untidy CSVs have all been standardized. The new database has September through November of 2017 and 2018.

I'm testing the database with a local API container today. I know of at least one bug, but it also affects the older database. I'll flag that as a separate issue; the logic I used to separate MAX and bus traces in the `stop_event` table is wrong. There are missing MAX traces in the `rail_passenger_stops` table and there are some MAX traces in the `bus_passenger_stops` table.

Testing update: database restores successfully. It uses about 27 GB of space. I'm not going to test the API until I've fixed the passenger stop bug and rerun `inspectdb`.
